bgpd/bgpd.c

```diff
@@ -3401,18 +3401,20 @@ static struct bgp *bgp_create(as_t *as, const char *name,
 	afi_t afi;
 	safi_t safi;
 
-	if (hidden) {
+	if (hidden)
 		bgp = bgp_old;
-		goto peer_init;
-	}
+	else
+		bgp = XCALLOC(MTYPE_BGP, sizeof(struct bgp));
 
-	bgp = XCALLOC(MTYPE_BGP, sizeof(struct bgp));
 	bgp->as = *as;
 	if (as_pretty)
 		bgp->as_pretty = XSTRDUP(MTYPE_BGP_NAME, as_pretty);
 	else
 		bgp->as_pretty = XSTRDUP(MTYPE_BGP_NAME, asn_asn2asplain(*as));
 
+	if (hidden)
+		goto peer_init;
```

Comment:
Here we don't have `bgp->peer` allocated, and `bgp->peer->cmp = (int (*)(void *, void *))peer_cmp;` will segfault after `goto peer_init;`. 

Also, we should respect AS notation...

Reply:
@askorichenko any comments on my concern above?